pkg/routeagent/controllers/route/route.go

```diff
@@ -449,9 +390,10 @@ func (r *Controller) processNextEndpoint() bool {
 		}
 
 		if endpoint.Spec.ClusterID != r.clusterID {
-			klog.V(6).Infof("Endpoint didn't match the cluster ID of this cluster")
+			klog.V(6).Infof("setting routes of endpoint object %s", name)
+			r.updateIptableRulesForInterclusterTraffic(endpoint.Spec.Subnets)
```

Comment:
Are we cleaning up the iptables rules when the endpoint is removed?

Reply:
We are not and we were not before either. That's some tech debt that we should tackle outside this work. What do you think?

Reply:
Sure